xdem/coreg.py

```diff
@@ -729,9 +876,7 @@ def fit_pts(
         ref_dem = reference_dem[ref_valid]
 
         if mask_high_curv:
-            planc, profc = xdem.terrain.get_terrain_attribute(
-                tba_dem, attribute=["planform_curvature", "profile_curvature"]
-            )
+            planc, profc = get_terrain_attribute(tba_dem, attribute=["planform_curvature", "profile_curvature"])
```

Comment:
This (together with `maxc`) will create three rasters, and then a fourth is created at `mask_hc`. We should probably have a "max_curvature" attribute for `get_terrain_attribute()`. But in the meantime, if you change this and the line below, this line will have three rasters in memory like before, but then two are garbage collected. Therefore, when `mask_hx` is defined, only two are in memory instead of four.

In code, could you change this to:

```python
maxc = np.maximum(np.abs(get_terrain_attribute(tba_dem, attribute=["planform_curvature", "profile_curvature"])), axis=0)
```
or add
```python
del planc
del profc
``` 
after `maxc` is created?

Reply:
Done :)